# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1327726455 28800
#      Fri Jan 27 20:54:15 2012 -0800
# Node ID 4f8d2931f8867ae2bc7a592940d62430ece398a2
# Parent  e2a14d1b4eaa2dfccdc183f2d4525c7864330e98
lines.m: Add new colormap corresponding to ColorOrder property.

* lines.m: New colormap file

* scripts/image/module.mk: Add lines.m to build system.

* image.txi: Add lines() to documentation.

* NEWS: Add lines() to list of new functions in 3.8

* aspell-octave.en.pws: Add "ColorOrder" to list of spelling exceptions.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,13 +1,14 @@
 Summary of important user-visible changes for version 3.8:
 ---------------------------------------------------------
 
  ** Other new functions added in 3.8.0:
 
+      lines
       rgbplot
 
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.4 and have been
     removed from Octave 3.8.
                                            
       autocor    dispatch              is_global   saveimage 
diff --git a/doc/interpreter/doccheck/aspell-octave.en.pws b/doc/interpreter/doccheck/aspell-octave.en.pws
--- a/doc/interpreter/doccheck/aspell-octave.en.pws
+++ b/doc/interpreter/doccheck/aspell-octave.en.pws
@@ -134,16 +134,17 @@ cmd
 cmember
 cntrl
 cof
 coffset
 colamd
 collectoutput
 colorbar
 colormap
+ColorOrder
 colperm
 Comint
 Commandline
 commentstyle
 ComplexEqn
 cond
 condest
 conformant
diff --git a/doc/interpreter/image.txi b/doc/interpreter/image.txi
--- a/doc/interpreter/image.txi
+++ b/doc/interpreter/image.txi
@@ -142,16 +142,18 @@ between 0 and 1.
 @DOCSTRING(gray)
 
 @DOCSTRING(hot)
 
 @DOCSTRING(hsv)
 
 @DOCSTRING(jet)
 
+@DOCSTRING(lines)
+
 @DOCSTRING(ocean)
 
 @DOCSTRING(pink)
 
 @DOCSTRING(prism)
 
 @DOCSTRING(rainbow)
 
diff --git a/scripts/image/lines.m b/scripts/image/lines.m
new file mode 100644
--- /dev/null
+++ b/scripts/image/lines.m
@@ -0,0 +1,54 @@
+## Copyright (C) 2012 Rik Wehbring
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {Function File} {@var{map} =} lines ()
+## @deftypefnx {Function File} {@var{map} =} lines (@var{n})
+## Create color colormap.  This colormap is composed of the list of colors
+## in the current axes "ColorOrder" property.  The default is blue,
+## green, red, cyan, pink, yellow, and gray.
+## The argument @var{n} must be a scalar.
+## If unspecified, the length of the current colormap, or 64, is used.
+## @seealso{colormap}
+## @end deftypefn
+
+function map = lines (n)
+
+  if (nargin == 0)
+    n = rows (colormap);
+  elseif (nargin == 1)
+    if (! isscalar (n))
+      error ("lines: argument must be a scalar");
+    endif
+  else
+    print_usage ();
+  endif
+
+  c = get (gca, "colororder");
+  nr = rows (c);
+  map = c(rem (0:(n-1), nr) + 1, :);
+
+endfunction
+
+
+%!demo
+%! ## Show the 'lines' colormap as an image
+%! image (1:64, linspace (0, 1, 64), repmat ((1:64)', 1, 64));
+%! axis ([1, 64, 0, 1], "ticy", "xy");
+%! colormap (lines (64));
+
diff --git a/scripts/image/module.mk b/scripts/image/module.mk
--- a/scripts/image/module.mk
+++ b/scripts/image/module.mk
@@ -19,16 +19,17 @@ image_FCN_FILES = \
   image/imagesc.m \
   image/imfinfo.m \
   image/imread.m \
   image/imshow.m \
   image/imwrite.m \
   image/ind2gray.m \
   image/ind2rgb.m \
   image/jet.m \
+  image/lines.m \
   image/ntsc2rgb.m \
   image/ocean.m \
   image/pink.m \
   image/prism.m \
   image/rainbow.m \
   image/rgb2hsv.m \
   image/rgb2ind.m \
   image/rgb2ntsc.m \
